test/cctest/test_aliased_buffer.cc

```diff
@@ -0,0 +1,220 @@
+
+#include "v8.h"
+#include "aliased_buffer.h"
+#include "node_test_fixture.h"
+
+using node::AliasedBuffer;
+
+class AliasBufferTest : public NodeTestFixture {
+ protected:
+  void SetUp() override {
+    NodeTestFixture::SetUp();
+  }
+
+  void TearDown() override {
+    NodeTestFixture::TearDown();
+  }
+};
+
+template<class NativeT>
+void CreateOracleValues(NativeT* buf, size_t count) {
+  for (size_t i = 0, j = count; i < count; i++, j--) {
+    buf[i] = static_cast<NativeT>(j);
+  }
+}
+
+template<class NativeT, class V8T>
+void WriteViaOperator(
+  AliasedBuffer<NativeT, V8T>& aliasedBuffer,  // NOLINT(runtime/references)
+  size_t size,
+  NativeT* oracle) {
+  // write through the API
+  for (size_t i = 0; i < size; i++) {
+    aliasedBuffer[i] = oracle[i];
+  }
+}
+
+template<class NativeT, class V8T>
+void WriteViaSetValue(
+  AliasedBuffer<NativeT, V8T>& aliasedBuffer,  // NOLINT(runtime/references)
+  size_t size, NativeT* oracle) {
+  // write through the API
+  for (size_t i = 0; i < size; i++) {
+    aliasedBuffer.SetValue(i, oracle[i]);
+  }
+}
+
+template<class NativeT, class V8T>
+void ReadAndValidate(
+  v8::Isolate* isolate,
+  v8::Local<v8::Context> context,
+  AliasedBuffer<NativeT, V8T>& aliasedBuffer,  // NOLINT(runtime/references)
+  size_t size,
+  NativeT* oracle) {
+  // read through the API
+  for (size_t i = 0; i < size; i++) {
+    NativeT v1 = (NativeT)aliasedBuffer[i];
+    NativeT v2 = (NativeT)aliasedBuffer.GetValue(i);
+    EXPECT_TRUE(v1 == oracle[i]);
+    EXPECT_TRUE(v2 == oracle[i]);
+  }
+
+  // validate size of JS Buffer
+  EXPECT_TRUE(aliasedBuffer.GetJSArray()->Length() == size);
+  EXPECT_TRUE(
+    aliasedBuffer.GetJSArray()->ByteLength() ==
+    (size * sizeof(NativeT)));
+
+  // validate operator * and GetBuffer are the same
+  EXPECT_TRUE(aliasedBuffer.GetNativeBuffer() == *aliasedBuffer);
+
+  // read through the JS API
+  for (size_t i = 0; i < size; i++) {
+    v8::Local<V8T> v8TypedArray = aliasedBuffer.GetJSArray();
+    v8::MaybeLocal<v8::Value> v = v8TypedArray->Get(context, i);
+    EXPECT_TRUE(v.IsEmpty() == false);
+    v8::Local<v8::Value> v2 = v.ToLocalChecked();
+    EXPECT_TRUE(v2->IsNumber());
+    // v8::MaybeLocal<v8::Number> v3 = v2->ToNumber();
+    // v8::Local<v8::Number> v4 = v3.ToLocalChecked();
+    // NativeT actualValue = *(reinterpret_cast<NativeT*>(&v4->Value()));
```

Comment:
I don't think this could work, you're taking a pointer to a stack-allocated return value and casting it around`.

`NativeT actualValue = static_cast<NativeT>(&v4->Value());`?

Reply:
I'll play around with that...

Reply:
OK, got this to work.  Think I was seeing seg fault earlier because I didn't pass `context` to `ToNumber`